test/edu/stanford/dlss/was/TestWasapiConnection.java

```diff
@@ -33,6 +37,24 @@ public void jsonQueryCallsExecute() throws IOException {
   }
 
   @Test
+  public void pagedJsonQuery_loopsThroughAndReturnsResponseList() throws IOException {
+    WasapiClient mockClient = mock(WasapiClient.class);
+    WasapiConnection spyConnection = spy(new WasapiConnection(mockClient));
+
+    WasapiResponse[] responses = {
+      mockResponseNotLast(),
+      mockResponseLast()
+    };
+    doReturn(responses[0]).when(spyConnection).jsonQuery(ORIG_QUERY_URL);
+    doReturn(responses[1]).when(spyConnection).jsonQuery(NEXT_URL);
+
+    List<WasapiResponse> respList = spyConnection.pagedJsonQuery(ORIG_QUERY_URL);
+    assertEquals("response list should have 2 entries", 2, respList.size());
+    assertEquals("response list's first entry should be first response", responses[0], respList.get(0));
+    assertEquals("response list's second entry should be second response", responses[1], respList.get(1));
+  }
+
```

Comment:
This looks good.  I would also like to see a test that is trying for a pagedJsonQuery, but it only returns a single page.  There is a fixture with that characteristic, fwiw.

Wondering also about a paged query request that returns no results (not that we tested that before ... and maybe that's not part of connection testing ...)